crates/revm/src/state_change.rs

```diff
@@ -329,9 +330,32 @@ where
 
     let withdrawal_requests = match result {
         ExecutionResult::Success { output, .. } => {
-            Vec::<Request>::decode(&mut output.into_data().as_ref()).map_err(|err| {
-                BlockValidationError::WithdrawalRequestsContractCall { message: err.to_string() }
-            })
+            let mut data = output.into_data();
+            let mut requests = Vec::with_capacity(data.len() / (20 + 48 + 8));
+
+            // Withdrawals are encoded as a series of withdrawal requests, each with the following
+            // format:
+            //
+            // +------+--------+--------+
+            // | addr | pubkey | amount |
+            // +------+--------+--------+
+            //    20      48        8
+            while data.has_remaining() {
```

Comment:
this loop will panic if data.len % 20 + 48 + 8 != 0

I assume this is not possible, but maybe not worth checking?